--- ../Tesla_Coil_MIDI_Synth/Knob.h	2024-11-10 14:40:13
+++ build/patched/Knob.h	2024-11-10 14:49:03
@@ -5,6 +5,7 @@
 #define BUTTON_DEBOUNCE 100 // ms
 #define ENCODER_ACCELERATION 60 // ms per decrement
 
+class Coil;
 class Synth;
 class LCD;
 class MIDI;
@@ -12,12 +13,13 @@
 
 class Knob {
 public:
-  Knob(Synth &synth, LCD &lcd, MIDI &midi, Audio &audio);
+  Knob(Coil &coil, Synth &synth, LCD &lcd, MIDI &midi, Audio &audio);
   
   void init();
   void update();
   
 private:
+  Coil &coil;
   Synth &synth;
   LCD &lcd;
   MIDI &midi;
@@ -31,8 +33,9 @@
   void updateEncoderAcceleration();
   void applyChange();
   
+public:
   // Encoder tracking and acceleration
-  static int8_t movement;
+  int8_t movement;
   static int32_t recentMovement;
   static unsigned long lastDecrement;
   static int32_t globalPos;
@@ -43,5 +46,5 @@
   static uint8_t encDir, encLutInd;
   
   // Static interrupt handler
-  static void enc();
+  void enc();
 };
