--- ../Tesla_Coil_MIDI_Synth/MIDI.h	2024-11-09 18:31:35
+++ build/patched/MIDI.h	2024-11-10 05:04:48
@@ -8,13 +8,14 @@
 // Uncomment to save MIDI base channel to flash
 // #define SAVE_BASE_MIDI
 
+class Coil;
 class Voices;
 class Synth;
 class LCD;
 
 class MIDI {
 public:
-  MIDI(Voices &voices, Synth &synth, LCD &lcd);
+  MIDI(Coil &coil, Voices &voices, Synth &synth, LCD &lcd);
   
   static const float midi2freq[];
   
@@ -148,6 +149,7 @@
   void process();
 
 private:
+  Coil &coil;
   Voices &voices;
   Synth &synth;
   LCD &lcd;
