--- ../Tesla_Coil_MIDI_Synth/Synth.h	2024-11-09 23:20:17
+++ build/patched/Synth.h	2024-11-10 05:04:48
@@ -25,12 +25,13 @@
 
 #define LUTSIZE 256
 
+class Coil;
 class Voices;
 class MIDI;
 
 class Synth {
 public:
-  Synth(Voices &voices, MIDI &midi);
+  Synth(Coil &coil, Voices &voices, MIDI &midi);
   
   uint8_t vol = DEFAULT_VOL;
   
@@ -39,6 +40,7 @@
   void update();
   
 private:
+  Coil &coil;
   Voices &voices;
   MIDI &midi;
   
